Sources/Fluent/Query/Join.swift

```diff
@@ -18,10 +18,34 @@ public struct Join {
     /// the local data
     public let foreign: Entity.Type
 
+    /// Direction the foreign key goes
+    public let child: ChildDirection
+
+    /// Indicates one of two directions the join can have.
+    /// 
+    /// Foreign: The foreign entity has a key
+    ///     that points to the local entity's
+    ///     primary key.
+    ///
+    /// Local: The local entity has a key
+    ///     that points to the foreign entity's
+    ///     primary key.
+    public enum ChildDirection {
+        case foreign
```

Comment:
Here too, can we make `foreignPointingToLocal` and `localPointingToForeign`. Do `local` and `foreign` have meaning here anymore? Should it just be `left` and `right` or something? The name local and foreign seem to imply a bunch of behavior that's not clear.